python/S3_PUBLIC_ACCESS_SETTINGS_FOR_ACCOUNT/S3_PUBLIC_ACCESS_SETTINGS_FOR_ACCOUNT.PY

```diff
@@ -0,0 +1,390 @@
+  | RestrictPublicBuckets  | Mandatory  | Block public and cross-account access to buckets that have public policies (True/False) |
+  | ---------------------- | ---------- | ----------------------------------------------------------------------------------------|
+Feature:
+  In order to: ensure that s3 account settings are being restricted to the appropriate level
+           As: a Security Officer
+       I want: to verify that the configuration of s3 account public access settings is correct.
+Scenarios:
+  Scenario 1:
+    Given: config parameters are defined for the config rule
+      And: At least 1 S3 config parameter does not match the corresponding value for the account
+     then: Return NON_COMPLIANT
+  Scenario 2:
+    Given: config parameters are defined for the config rule
+      And: All S3 config parameters match the corresponding value for the account
+     then: Return COMPLIANT
+'''
+
+import json
+import datetime
+import os
+import os.path
+import sys
+envLambdaTaskRoot = os.environ["LAMBDA_TASK_ROOT"]  
+sys.path.insert(0,envLambdaTaskRoot+"/boto3-1-9-82") #hack to change the version of boto
+import boto3
+import botocore
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::::Account'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+#############
+# Main Code #
+#############
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+  
+    client = boto3.client('s3control')
+    AWS_ACCOUNT_ID = boto3.client('sts').get_caller_identity().get('Account')
+    response = client.get_public_access_block(AccountId=AWS_ACCOUNT_ID)
+    evaluations = []
+    annotationbuilder = ''
+
+    if (response['PublicAccessBlockConfiguration']['BlockPublicAcls'] == valid_rule_parameters['BlockPublicAcls']) \
+    and (response['PublicAccessBlockConfiguration']['IgnorePublicAcls'] == valid_rule_parameters['IgnorePublicAcls']) \
+    and (response['PublicAccessBlockConfiguration']['BlockPublicPolicy'] == valid_rule_parameters['BlockPublicPolicy']) \
+    and (response['PublicAccessBlockConfiguration']['RestrictPublicBuckets'] == valid_rule_parameters['RestrictPublicBuckets']):
+        annotationbuilder = 'BlockPublicAcls:' + str(response['PublicAccessBlockConfiguration']['BlockPublicAcls']) + ' ' \
+        'IgnorePublicAcls:' + str(response['PublicAccessBlockConfiguration']['IgnorePublicAcls']) + ' ' \
+        'BlockPublicPolicy:' + str(response['PublicAccessBlockConfiguration']['BlockPublicPolicy']) + ' ' \
+        'RestrictPublicBuckets:' + str(response['PublicAccessBlockConfiguration']['RestrictPublicBuckets'])
+        evaluations.append(build_evaluation(AWS_ACCOUNT_ID, 'COMPLIANT', event, annotation=annotationbuilder))

```

Comment:
No annotation needed, when COMPLIANT

Reply:
Removed annotation for compliant if statement